### ISSUE TITLE
Library does not allow to extract on mask

### ISSUE THREAD

**Issue by gmihaila** [thunlp/OpenPrompt#116]
There is no way to extract the mask prediction... the [extract_at_mask)](https://thunlp.github.io/OpenPrompt/modules/base.html?highlight=mask#openprompt.pipeline_base.PromptForClassification.extract_at_mask) is only implemented for `PromptForClassification` which only outputs the prediction of classification layer... 

Is there a reason why we can't see the predicted masked tokens? The documentation has very good intentions but it misses a lot of the library functionality and does not provide clear examples. Hope someone can shed some light here. I just want to see what was the predicted mask token.

**Comment by Achazwl** [comment]
extract_at_mask is only implemented for PromptForClassification but does not output the prediction of the classification layer. With our TokenizerWrapper and PromptDataloader, key `loss_ids` with be added to the inputs, which indicates the `<mask>` position, or for T5 indicates the position after `<extra_id_0>` in decoder. The extract_at_mask will read from the loss_ids to extract. You can refer to our tutorial folder in this repo, for more clear examples.

**Comment by gmihaila** [comment]
@Achazwl Thank you for the reply! 

It is very hard to follow your tutorials if they don't have any comments or describing what you are trying to do in those scripts besides besides the name of the scripts. I started following the code and logic but it takes a lot of my time... It would be nice to have better documented scripts in your tutorial.

It looks like great work what you did here but it takes a lot of work to understand what your implementation is doing without a proper documentation.
